lib/clearance/password_reset_token.rb

```diff
@@ -0,0 +1,43 @@
+module Clearance
+  class PasswordResetToken
+    def self.generate_for(user)
+      token = Clearance.configuration.message_verifier.generate([
+        user.id,
+        Digest::MD5.hexdigest(user.encrypted_password),
```

Comment:
Agree, avoid MD5 for any new code and use SHA256 instead. 

I'm wondering why this PR is only using signed messages and not encrypted and signed ones. An example of the latter is JSON Web Tokens (https://jwt.io/), those use HMAC-SHA256 and their usecase is nearly similar to this one.